.github/workflows/kind.yml

```
@@ -38,96 +30,196 @@ jobs:
         with:
           go-version-file: go.mod
 
-      - name: Verify modules
+      - name: Format, Lint, and Generate
         run: |
           go mod verify
-
-      - name: Verify format
-        run: |
           make fmt
           git diff --exit-code
           make lint
-
-      - name: Verify deploy/operator.yaml
-        run: |
           make generate
           make generate-operator-yaml
           git diff --exit-code
 
-      - name: Unit and Integration Tests
+      - name: Unit Tests
+        run: |
+          make test-coverage
```

Looks like there's a `tee` in the other command that's missed here.
```suggestion
          make test-coverage | tee report_unit.json
```

---

Take back the takeback--looks like we reference the JSON in the `sonar-project.properties`, so it's probably safer to leave it in place.

---

Good eye though! The target makes a `coverage_unit.out` file, which is saved and eventually sent to SonarCloud.

I just ran the old `make test-coverage | tee report_unit.json` command and it did not produce a valid JSON file... and it's fundamentally different than the report_e2e.json file I have from a different set of tests, so I think this file wasn't being used.

---

I'll do some research on the `sonar.go.tests.reportPaths`, I feel like the current file might not really be what sonarcloud wants

---

You're right--we weren't even uploading it 😆 
https://github.com/stolostron/config-policy-controller/blob/426c9d0c30b8de7d257403182313c799fe79bad4/.github/workflows/kind.yml#L120-L123

---

I suspect it only uses it to parse for a failing test, which would be blocking in the PR for us anyway.

---

Yeah, the scan is showing them as not found :laughing: https://github.com/stolostron/config-policy-controller/actions/runs/10354923804/job/28661334480#step:10:463
```
15:35:15.224 ERROR: Test report can't be loaded, file not found: '/github/workspace/report.json', ignoring this file.
15:35:15.224 ERROR: Test report can't be loaded, file not found: '/github/workspace/report_e2e.json', ignoring this file.
15:35:15.224 ERROR: Test report can't be loaded, file not found: '/github/workspace/report_unit.json', ignoring this file.
```

---

Ok, I think the ginkgo json report we have at `report_e2e.json` is a very specific format to ginkgo, it's not clear if Sonar would parse it. It's definitely different than the JSON stream format that `go test` produces by default, which is kind of what we have in `report_unit.json`, except there are some additional non-json lines when it's `tee`'d:
```
Checking installation of sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
Installing Kubebuilder
KUBEBUILDER_ASSETS="/home/jkulikau/.local/share/kubebuilder-envtest/k8s/1.29.1-linux-amd64" go test -json -cover -covermode=atomic -coverprofile=coverage_unit.out `go list ./... | grep -v test/e2e`
{"Time":"2024-08-13T10:30:30.854198776-04:00","Action":"start","Package":"open-cluster-management.io/config-policy-controller"}
{"Time":"2024-08-13T10:30:30.859181709-04:00","Action":"output","Package":"open-cluster-management.io/config-policy-controller","Output":"\topen-cluster-management.io/config-policy-controller\t\tcoverage: 0.0% of statements\n"}
...
```

Sonar probably would understand the go-default JSON-stream if we cleaned it up, but our unit tests are... not super interesting, especially if we ensure they all pass before we submit the report. We have the coverage information correctly submitted, and I think that's the main thing we want.